pkg/provider/azure_vmss.go

```diff
@@ -1674,5 +1674,10 @@ func (ss *ScaleSet) EnsureBackendPoolDeletedFromVMSets(vmssNamesMap map[string]b
 
 // GetAgentPoolVMSetNames returns all VMSS names according to the nodes
 func (ss *ScaleSet) GetAgentPoolVMSetNames(nodes []*v1.Node) (*[]string, error) {
-	return ss.getAgentPoolScaleSets(nodes)
+	vmssNames, err := ss.getAgentPoolScaleSets(nodes)
```

Comment:
we probably need to get vmSet list node by node? there're chances some nodes are VMSS, while others are VMAS